plugins/Jobs.cpp

```diff
@@ -791,10 +788,12 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
         if (!retriableJobs.empty()) {
             SINFO("Updating jobs with retryAfter");
             for (auto job : retriableJobs) {
+                string currentTime = SCURRENT_TIMESTAMP();
+                string nextRunDateTime = _constructNextRunDATETIME(job["nextRun"], job["lastRun"] != "" ? job["lastRun"] : job["nextRun"], job["repeat"]);
                 string updateQuery = "UPDATE jobs "
                                      "SET state='RUNQUEUED', "
-                                         "lastRun=" + SCURRENT_TIMESTAMP() + ", "
-                                         "nextRun=DATETIME(" + SCURRENT_TIMESTAMP() + ", " + SQ(job["retryAfter"]) + ") "
+                                         "lastRun=" + currentTime + ", "
+                                         "nextRun=MIN(DATETIME(" + currentTime + ", " + SQ(job["retryAfter"]) + "), COALESCE(NULLIF(" + (nextRunDateTime != "" ? nextRunDateTime : "''") + ", ''), 'z'))" + " "
```

Comment:
`NULLIF` is necessary because `nextRunDateTime` comes out as empty string (not null) and that is always the `MIN`, so we need it for the COALESCE to work correctly.